test/test_nestedtensor.py

```diff
@@ -7818,131 +7822,539 @@ def __torch_dispatch__(self, func, types, args=..., kwargs=None):
+FORWARD_FAILURES = [
+    # not implemented
+    XFailRule(
+        error_type=NotImplementedError,
+        match_fn=lambda device, dtype, op, sample: op.full_name
+        in {
+            # unary
+            "nn.functional.celu",
+            "nn.functional.elu",
+            "nn.functional.hardshrink",
+            "nn.functional.hardsigmoid",
+            "nn.functional.hardtanh",
+            "nn.functional.logsigmoid",
+            "nn.functional.mish",
+            "nn.functional.relu6",
+            "nn.functional.rrelu",
+            "nn.functional.selu",
+            "nn.functional.softplus",
+            "nn.functional.softshrink",
+            "nn.functional.threshold",
+            "rad2deg",
+            # binary
+            "__rsub__",
+            "complex",
+            "floor_divide",
+            "polar",
+            "rsub",
+            # reduction
+            "count_nonzero",
+            "linalg.vector_norm",
+            "nansum",
+            "std",
+            "std.unbiased",
+            "var",
+            "var.unbiased",
+        },
+        name="not_implemented",
+    ),
+    # expected: masked ops don't support jagged layout
+    XFailRule(
+        error_type=ValueError,
+        error_msg="expects strided",
+        match_fn=lambda device, dtype, op, sample: op.full_name
+        in {
+            "masked.amax",
+            "masked.amin",
+            "masked.argmax",
+            "masked.argmin",
+            "masked.logsumexp",
+            "masked.mean",
+            "masked.norm",
+            "masked.prod",
+            "masked.std",
+            "masked.sum",
+            "masked.var",
+        },
+        name="no_masked_jagged_support",
+    ),
     # Need to adjust sample input func to pass the right thing
-    "nn.functional.prelu",
-    # TypeError: fill() received an invalid combination of arguments
-    # got (NestedTensor), but expected one of:
-    # * (Tensor input, Tensor value)
-    # * (Tensor input, Number value)
-    "fill",
-    # RuntimeError: unsupported tensor layout: Jagged
-    "jiterator_binary",
-    "jiterator_binary_return_by_ref",
-    "jiterator_unary",
-    # RuntimeError: "jagged_to_padded_dense" not implemented for 'Bool'
-    "nanmean",
-}
+    XFailRule(
+        error_type=TypeError,
+        error_msg="missing 1 required positional arguments",
+        match_fn=lambda device, dtype, op, sample: op.full_name
+        == "nn.functional.prelu",
+        name="invalid_prelu_sample_input_func",
+    ),
+    # Op doesn't support lengths being present
+    XFailRule(
+        error_type=ValueError,
+        error_msg="expected input to be a contiguous jagged layout NestedTensor",
+        match_fn=lambda device, dtype, op, sample: (
+            op.full_name == "nn.functional.linear" and sample.input._lengths is not None
+        ),
+        name="no_linear_noncontig_holes_support",
+    ),
+    # Some kinda reduction bug that needs to be fixed!
+    XFailRule(
```

Comment:
It feels like this top most section of xfails is like the "expected" section, is it worth moving this bug below somewhere (or does it fit better here for other reasons?)

Reply:
Don't want this type of stuff to block this PR, but re https://github.com/pytorch/pytorch/pull/138370/files#r1834994353, more of just curious of what the plan of organization is.

Reply:
(see discussion below on xfails, resolving this one)